--- lustre-release-f7948c6/config/lustre-build.m4.orig	2024-11-27 23:19:27.000000000 +0100
+++ lustre-release-f7948c6/config/lustre-build.m4	2025-04-15 05:42:27.332508008 +0200
@@ -154,6 +154,7 @@ AC_ARG_ENABLE([modules],
 	])
 AC_MSG_RESULT([$enable_modules ($target_os)])
 
+LC_PROG_LINUX
 AS_IF([test "x$enable_modules" = xyes], [
 	AS_IF([test "x$FLEX" = "x"], [AC_MSG_ERROR([flex package is required to build kernel modules])])
 	AS_IF([test "x$BISON" = "x"], [AC_MSG_ERROR([bison package is required to build kernel modules])])
@@ -167,7 +168,6 @@ AS_IF([test "x$enable_modules" = xyes],
 			LIBCFS_PROG_LINUX
 			LN_PROG_LINUX
 			AS_IF([test "x$enable_server" != xno], [LB_EXT4_SRC_DIR])
-			LC_PROG_LINUX
 
 			# Run any parallel compile tests
 			LB_PROG_LINUX_SRC
@@ -513,7 +513,6 @@ AC_ARG_ENABLE([server],
 # before running, so until LB_CONFIG_MODULES can be reorganized, we
 # call it here.
 LB_CONFIG_MODULES
-AS_IF([test x$enable_modules = xno], [enable_server=no])
 LB_CONFIG_LDISKFS
 LB_CONFIG_ZFS
 
@@ -611,10 +610,10 @@ elif test x$enable_gss = xno ; then
 	RPMBINARGS="$RPMBINARGS --without gss"
 	AC_SUBST(ENABLE_GSS, no)
 fi
-if test x$enable_crypto = xyes ; then
+if test x"$enable_crypto" != xno ; then
 	RPMBINARGS="$RPMBINARGS --with crypto"
 	AC_SUBST(ENABLE_CRYPTO, yes)
-elif test x$enable_crypto = xno ; then
+elif test x"$enable_crypto" = xno ; then
 	RPMBINARGS="$RPMBINARGS --without crypto"
 	AC_SUBST(ENABLE_CRYPTO, no)
 fi
--- lustre-release-cac870c/lustre/autoconf/lustre-core.m4.orig	2023-12-23 04:15:24.000000000 +0100
+++ lustre-release-cac870c/lustre/autoconf/lustre-core.m4	2024-05-20 20:26:24.607425373 +0200
@@ -286,7 +286,6 @@ AC_ARG_ENABLE([gss_keyring],
 AC_MSG_RESULT([$enable_gss_keyring])
 AS_IF([test "x$enable_gss_keyring" != xno], [
 	LB_CHECK_CONFIG_IM([KEYS], [], [
-		gss_keyring_conf_test="fail"
 		AC_MSG_WARN([GSS keyring backend requires that CONFIG_KEYS be enabled in your kernel.])])
 
 	AC_CHECK_LIB([keyutils], [keyctl_search], [], [
@@ -334,7 +333,7 @@ AC_DEFUN([LC_KEY_TYPE_INSTANTIATE_2ARGS]
 AC_DEFUN([LC_CONFIG_SUNRPC], [
 LB_CHECK_CONFIG_IM([SUNRPC], [], [
 	AS_IF([test "x$sunrpc_required" = xyes], [
-		AC_MSG_ERROR([
+		AC_MSG_WARN([
 
 kernel SUNRPC support is required by using GSS.
 ])
@@ -4081,7 +4081,7 @@ AS_IF([test "x$enable_crypto" = xin-kern
 	      AC_DEFINE(HAVE_LUSTRE_CRYPTO, 1, [Enable Lustre client crypto via in-kernel fscrypt])], [
 	      AC_MSG_ERROR([Lustre client crypto cannot be enabled via in-kernel fscrypt.])
 	      enable_crypto=no])],
-	[AS_IF([test "x$has_is_encrypted" = xyes], [
+	[AS_IF([test "x$enable_crypto" != xno], [
 	      AC_DEFINE(HAVE_LUSTRE_CRYPTO, 1, [Enable Lustre client crypto via embedded llcrypt])
 	      AC_DEFINE(CONFIG_LL_ENCRYPTION, 1, [embedded llcrypt])
 	      AC_DEFINE(HAVE_FSCRYPT_DUMMY_CONTEXT_ENABLED, 1, [embedded llcrypt uses llcrypt_dummy_context_enabled()])
--- lustre-release-f7948c6/config/lustre-build-linux.m4.orig	2025-04-15 05:38:14.350545196 +0200
+++ lustre-release-f7948c6/config/lustre-build-linux.m4	2025-04-15 05:38:20.513845139 +0200
@@ -269,7 +269,7 @@ if test "$DEFAULT_LINUX" = "/lib/modules
 else
 	PATHS="/usr/src/linux-headers-$(uname -r)"
 fi
-PATHS+=" $DEFAULT_LINUX"
+PATHS="$PATHS $DEFAULT_LINUX"
 for DEFAULT_LINUX_OBJ in $PATHS; do
 	AS_IF([readlink -q -e $DEFAULT_LINUX_OBJ >/dev/null], [break])
 done
--- lustre-release-f7948c6/config/lustre-build-zfs.m4.orig	2024-11-27 23:19:27.000000000 +0100
+++ lustre-release-f7948c6/config/lustre-build-zfs.m4	2025-04-15 05:39:12.490230226 +0200
@@ -783,7 +783,7 @@ your distribution.
 				[Have zfs_refcount.h])
 		])
 		old_EXTRA_KCFLAGS=$EXTRA_KCFLAGS
-		EXTRA_KCFLAGS+=" -Werror"
+		EXTRA_KCFLAGS="$EXTRA_KCFLAGS -Werror"
 		dnl #
 		dnl # ZFS 2.2.0 nvpair now returns and expects constant args
 		dnl #
